authorization_test.go

```diff
@@ -0,0 +1,105 @@
+package certauth_test
+
+import (
+	"testing"
+
+	. "github.com/pantheon-systems/go-certauth"
+)
+
+func TestAuthValidateOU(t *testing.T) {
+	// Tests that OU validation works as expected
+	tests := []struct {
+		AllowedOUs []string
+		ActualOUs  []string
+		IsAllowed  bool
+	}{
+		{[]string{}, []string{"endpoint"}, true},
+		{[]string{"endpoint"}, []string{"endpoint"}, true},
+		{[]string{"endpoint"}, []string{"site"}, false},
+		{[]string{"endpoint"}, []string{""}, false},
+		{[]string{"endpoint", "titan"}, []string{"site"}, false},
+		{[]string{"endpoint", "titan"}, []string{"titan"}, true},
+	}
+
+	for _, tc := range tests {
+		check := AllowSpecificOUandCNs{OUs: tc.AllowedOUs, CNs: nil}
+		_, err := check.CheckAuthorization(tc.ActualOUs, "")
+
+		if err != nil && tc.IsAllowed {
+			t.Errorf("Expected AllowedOUs (%v) and ActualOUs (%v) to pass validation, but it failed: err: %s", tc.AllowedOUs, tc.ActualOUs, err)
+		}
+
+		if err == nil && !tc.IsAllowed {
+			t.Errorf("Expected AllowedOUs (%v) and ActualOUs (%v) to fail validation, but it passed.", tc.AllowedOUs, tc.ActualOUs)
+		}
+	}
+}
+
+func TestAuthValidateCN(t *testing.T) {
+	// Tests that CN validation works as expected
+	tests := []struct {
+		AllowedCNs []string
+		ActualCN   string
+		IsAllowed  bool
+	}{
+		{[]string{}, "site1", true},
+		{[]string{"site1"}, "site1", true},
+		{[]string{"site1"}, "site", false},
+		{[]string{"site1"}, "", false},
+		{[]string{"site1", "site2"}, "site1", true},
+		{[]string{"site1", "site2"}, "site2", true},
+		{[]string{"site1", "site2"}, "site3", false},
+	}
+
+	for _, tc := range tests {
+		check := AllowSpecificOUandCNs{OUs: nil, CNs: tc.AllowedCNs}
+		_, err := check.CheckAuthorization([]string{""}, tc.ActualCN)
+
+		if err != nil && tc.IsAllowed {
+			t.Errorf("Expected AllowedCNs (%v) and ActualCN (%v) to pass validation, but it failed: err: %s", tc.AllowedCNs, tc.ActualCN, err)
```

Comment:
Having difficulty figuring out when you are and are not ok with long lines. Seems strange to newline function parameters and leave this as is.

Reply:
I wrap these by hand cause auto-wrap in VIM tends to get undone by `go fmt`. Occasionally I miss a line :shrug: